lib/elastic_apm/middlewares/grape.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Source: https://github.com/elastic/apm-agent-ruby/blob/master/lib/elastic_apm/middleware.rb
+module ElasticAPM
+  module Middlewares
+    # @api private
+    class Grape
+      def initialize(app)
+        @app = app
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def call(env)
+        begin
+          transaction = ElasticAPM.transaction transaction_name(env), 'app',
+            context: ElasticAPM.build_context(env)
+
+          resp = @app.call env
+
+          transaction.submit(resp[0], headers: resp[1]) if transaction
+        rescue InternalError
+          raise # Don't report ElasticAPM errors
+        rescue ::Exception => e
+          ElasticAPM.report(e, handled: false)
+          transaction.submit(500) if transaction
```

Comment:
As i know, if you use `rescue_from` inside your grape API you will get 500 status code, because `rescue_from` processed after middleware execution.

For example:

```rb
require 'grape'

class SomeMiddleware
  def initialize(app)
    @app = app
  end

  def call(env)
    @app.call(env)
  rescue StandardError
    puts 'Unexpected behavior'
  end
end

class API < Grape::API
  use SomeMiddleware

  rescue_from StandardError do |e|
    puts 'Expected behavior'
  end

  get do
    puts 'API CALL'
    raise StandardError
  end
end

run API
```

You will see:

```
API CALL
Unexpected behavior
```